src/target/riscv/riscv-013.c

```diff
@@ -2413,41 +2410,41 @@ static int write_memory_abstract(struct target *target, target_addr_t address,
 	LOG_DEBUG("writing %d words of %d bytes from 0x%" TARGET_PRIxADDR, count,
 			  size, address);
 
-	// Convert the size (bytes) to width (bits)
+	/* Convert the size (bytes) to width (bits) */
 	unsigned width = size << 3;
 	if (width > 64) {
-		// TODO: Add 128b support if it's ever used. Involves modifying
-		//       read/write_abstract_arg() to work on two 64b values.
+		/* TODO: Add 128b support if it's ever used. Involves modifying
+				 read/write_abstract_arg() to work on two 64b values. */
 		LOG_ERROR("Unsupported size: %d bits", width);
 		return ERROR_FAIL;
 	}
 
-	// Create the command (physical address, postincrement, write)
+	/* Create the command (physical address, postincrement, write) */
 	uint32_t command = access_memory_command(target, false, width, true, true);
 
-	// Execute the writes
-	const uint8_t* p = buffer;
+	/* Execute the writes */
+	const uint8_t *p = buffer;
 	bool updateaddr = true;
 	for (uint32_t c = 0; c < count; c++) {
-		// Move data to arg0
-		riscv_reg_t value = *(riscv_reg_t*)p;
+		/* Move data to arg0 */
+		riscv_reg_t value = *p;
```

Comment:
Does that work? p is uint8_t*, so dereferencing it will just get you a byte and not a full value, right?

Reply:
Oops, yeah, sorry. Was chasing a clang error about ignoring alignment, and forgot that I intended this to grab a full word of data even if fewer bits are written. Will just use a memcpy instead.